be/src/vec/data_types/data_type_struct.h

```diff
@@ -68,6 +68,19 @@ class DataTypeStruct final : public IDataType {
 public:
     static constexpr bool is_parametric = true;
 
+    // Add helper function for case-insensitive comparison
+    static bool case_insensitive_equals(const String& a, const String& b) {
```

Comment:
find some util  in boost or std or other thirdparty,not manual write insensitive compare